pkg/agent/agent_windows.go

```diff
@@ -150,14 +150,14 @@ func (i *Initializer) prepareOVSBridge() error {
 	if err = util.SetAdapterMACAddress(brName, &uplinkNetConfig.MAC); err != nil {
 		return err
 	}
-	// Remove existing IP addresses to avoid a candidate error of "Instance MSFT_NetIPAddress already exists" when
-	// adding it on the adapter.
-	if err = util.RemoveIPv4AddrsFromAdapter(brName); err != nil {
-		return err
-	}
 	// TODO: Configure IPv6 Address.
 	if err = util.ConfigureInterfaceAddressWithDefaultGateway(brName, uplinkNetConfig.IP, uplinkNetConfig.Gateway); err != nil {
-		return err
+		if strings.Contains(err.Error(), "Instance MSFT_NetIPAddress already exists") {
+			err = nil
+			klog.V(4).Infof("Address: %s already exists on interface %s", uplinkNetConfig.IP.String(), brName)
```

Comment:
Could you explain to me when this can happen?
It is for IP already exists on br-int, or also when the IP exists on another adapter of the host?

Reply:
@jianjuns :
> Could you explain to me when this can happen?
0. The IP is on Ethernet0 at first.
1. Set the Ethernet0 as uplink and the Ethernet0 wil have no IP. The corresponding `MSFT_NetIPAddress ` item will also be removed.
3. Add `br-int` to OVS and set the MAC same as `Ethernet0`.
    -  If the DHCP is enabled. `br-int` may get the same IP, which will cause `Instance MSFT_NetIPAddress already exists` when we try to set the IP later.

> It is for IP already exists on br-int, or also when the IP exists on another adapter of the host?
It's for IP already exists on br-int. Actually the IP has change exist on another adapter. But I think it's almost impossible unless some other process set the IP on another adapter during these steps. So I didn't put too much check here.

Change log info to: `klog.V(4).Infof("Address: %s already exists when configuring IP on interface %s", uplinkNetConfig.IP.String(), brName)`. It should be more accurate.

Reply:
Could be better if you add the explanation (DHCP etc.) to the code comments.